### ISSUE TITLE
Ce

### ISSUE THREAD

**Issue by papanito** [spantaleev/matrix-docker-ansible-deploy#947]

**Comment by spantaleev** [comment]
Somewhat related to #80?

**Comment by papanito** [comment]
@spantaleev yeah somewhat. Collections are quite new and I think it's easier to achieve as to my understanding one only has to create a specific structure

>Collections follow a simple data structure. None of the directories are required unless you have specific content that belongs in one of them. A collection does require a galaxy.yml file at the root level of the collection. This file contains all of the metadata that Galaxy and other tools need in order to package, build and publish the collection:

```
collection/
├── docs/
├── galaxy.yml
├── meta/
│   └── runtime.yml
├── plugins/
│   ├── modules/
│   │   └── module1.py
│   ├── inventory/
│   └── .../
├── README.md
├── roles/
│   ├── role1/
│   ├── role2/
│   └── .../
├── playbooks/
│   ├── files/
│   ├── vars/
│   ├── templates/
│   └── tasks/
└── tests/
```

**Comment by papanito** [comment]
Which, if I see correctly, also eliminates the need to have the roles independent as stated in #80

**Comment by papanito** [comment]
If you tell me this is something to try, I might have a look. Never made a collection, but definitively something I would like to help

**Comment by papanito** [comment]
@spantaleev here an example of how the repos should be restructured to be an ansible collection

https://github.com/papanito/matrix-docker-ansible-deploy/tree/papanito/ansible-collection

So far I only made the necessary changes to be able to upload it to ansible galaxy, still needs testing. Same goes for your playbook - it was moved to `playbooks` folder but did not test it yet.

Also the `README` should probably be updated. But before I go on I would like to hear your opinion/feedback.

**Comment by dev-zero** [comment]
This would be really nice. Are there plans to avoid breaking things for current users?

**Comment by papanito** [comment]
The playbooks may need some adjustments but I believe it should be doable without breaking the playbooks. Ultimately the roles are the same just they will just be belong to a collection...

**Comment by spantaleev** [comment]
Closing this as part of a backlog cleanup. Restructuring the playbook into an Ansible collection would be a large architectural change, and in the years since this was filed nobody has planned or started such work. The playbook has meanwhile been moving in a different direction: reusable roles get extracted to standalone repositories (mostly under the mother-of-all-self-hosting organization) and are pulled in via ansible-galaxy, which addresses much of the reusability motivation behind a collection. If a contributor wants to champion a collection restructure and demonstrate its benefits, this can be revived.
